src/agent.ts

```diff
@@ -1,68 +1,45 @@
-export const TETHER_ADDRESS = "0xdAC17F958D2ee523a2206206994597C13D831ec7";
-export const TETHER_DECIMALS = 6;
-let findingsCount = 0;
-
-const handleTransaction: HandleTransaction = async (
-  txEvent: TransactionEvent
-) => {
-  const findings: Finding[] = [];
-
-  // limiting this agent to emit only 5 findings so that the alert feed is not spammed
-  if (findingsCount >= 5) return findings;
-
-  // filter the transaction logs for Tether transfer events
-  const tetherTransferEvents = txEvent.filterLog(
-    ERC20_TRANSFER_EVENT,
-    TETHER_ADDRESS
-  );
-
-  tetherTransferEvents.forEach((transferEvent) => {
-    // extract transfer event arguments
-    const { to, from, value } = transferEvent.args;
-    // shift decimals of transfer value
-    const normalizedValue = value.div(10 ** TETHER_DECIMALS);
-
-    // if more than 10,000 Tether were transferred, report it
-    if (normalizedValue.gt(10000)) {
-      findings.push(
-        Finding.fromObject({
-          name: "High Tether Transfer",
-          description: `High amount of USDT transferred: ${normalizedValue}`,
-          alertId: "FORTA-1",
-          severity: FindingSeverity.Low,
-          type: FindingType.Info,
-          metadata: {
-            to,
-            from,
-          },
-        })
-      );
-      findingsCount++;
+import { providers } from "ethers";
+import { Finding, HandleTransaction, TransactionEvent, getEthersProvider } from "forta-agent";
+import { createFinding } from "./finding";
+import { IUNISWAPV3POOL, SWAP_EVENT } from "./constants";
+import { getPoolValues, getFactoryContract, createTwoAddress } from "./helper";
+
+export function provideHandleTransaction(
+  provider: providers.Provider,
+  SWAP_EVENT: string[],
+  IUNISWAPV3POOL: string[]
+): HandleTransaction {
+  return async (txEvent: TransactionEvent): Promise<Finding[]> => {
+    const findings: Finding[] = [];
+
+    //Get the factory contract to use later when we call getPool func to verify pool address
+    const factoryContract = getFactoryContract(provider);
+    const bn = txEvent.blockNumber;
+
+    // Filter out any swap logs from the blockchain that match our SWAP_EVENT
+    const swapLogs = txEvent.filterLog(SWAP_EVENT);
+    for (const log of swapLogs) {
+      //We wrap this in a try/catch as any invalid uniswap pools will error out
+      try {
+        //Call getPoolValues to get the token addresses and fee from the pool contract
+        //if this is a valid pool contract it will return the vals
+        const poolVal = await getPoolValues(log.address, IUNISWAPV3POOL, provider, bn);
+
+        //once we receive the poolVals we use create2 to get the pool address
+        //if the pool address from the event log matches this pool address we have a valid uniswap pool
+        //We can then create a finding and push it to our findings array
+        const poolAddress = await createTwoAddress(poolVal, factoryContract);
```

Comment:
```suggestion
        const poolAddress = createTwoAddress(poolVal, factoryContract);
```

This `await` is not needed